### PR TITLE
ref(slack): Remove slack-v2 creds and usage

### PR DESCRIPTION
**Pull request by MeredithAnya**
Once https://github.com/getsentry/sentry/pull/23545 is merged and deployed, we will have removed basically all the functionality for slack workspace apps. The last piece is the app itself, and it's credentials. 

**IMPORTANT**: This PR should only be merged **AFTER** we have updated our slack-v2 credentials to replace our slack credentials. 

We started out with one slack app, our now deprecated workspace app. The credentials for this app are:
```python
"slack.client-id"
"slack.client-secret"
"slack.verification-token"
# added later when we introduced support for signing secrets
"slack.signing-secret"
```

We then introduced a new Slack integration, the bot app. The credentials for that app are:
```python
"slack-v2.client-id"
"slack-v2.client-secret"
"slack.signing-secret"
```

Now there is actually a third type of app which is a older bot app, this kind of bot app may still be used by self hosted sentry users, and may be using the `slack.verification-token` instead of the signing secret (though they should probably update their app to use the secret). But anyway this is the reason we need to keep that option, although sentry SaaS will not need it. 

mini clean up: we don't use the option `slack.legacy-app` anymore, so getting rid of it

Side Note: I'm mocking out the signing secret check in the endpoint tests because I couldn't get the payloads to match for some reason,I want to come back to this, but it isn't any worse that what it was before because these tests used the verification token which we don't use anymore

### CODE REVIEW

**Review by scefali — COMMENTED** [review]
LGTM, I'll approve once the other PR is merged just so we don't accidentally merge this first

**Comment by MeredithAnya** [comment]
Confirming here that the following credentials have been updated: 

* `slack.client-id` is set [here](https://github.com/getsentry/getsentry/blob/f4dffb91b875942976cae7f882eb76565984d88b/getsentry/conf/settings/prod.py#L620) 
* `slack.client-secret` uses the env var `SLACK_CLIENT_SECRET` which was updated by @b1naryth1ef 
* `slack.signing-secret` uses the env var `SLACK_SIGNING_SECRET` which was updated by @b1naryth1ef 

The `slack.verification-token` uses the env var `SLACK_VERIFICATION_TOKEN` and was not changed since we don't use it. (we always look for the signing secret first)

**Review by scefali — APPROVED** [review]

File: tests/sentry/integrations/slack/test_action_endpoint.py
Comment (MeredithAnya) [inline tests/sentry/integrations/slack/test_action_endpoint.py:432]: can't seem to get the signatures to match when I actually pass in the correct headers :(

**Pull request merged**
